From bf736eb9491042b119d18c1401f7563b400e1e59 Mon Sep 17 00:00:00 2001
Message-Id: <bf736eb9491042b119d18c1401f7563b400e1e59.1462873074.git.pmatilai@redhat.com>
From: Panu Matilainen <pmatilai@redhat.com>
Date: Tue, 10 May 2016 12:35:43 +0300
Subject: [PATCH] qede: fix build with gcc >= 6.0

With gcc >= 6.0, qede base driver fails to build with:
drivers/net/qede/base/ecore_cxt.c: In function 'ecore_cdu_init_common':
cc1: error: left shift of negative value [-Werror=shift-negative-value]

Since the base drivers are untouchable, work around by disabling
the warning.

Signed-off-by: Panu Matilainen <pmatilai@redhat.com>
---
 drivers/net/qede/Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/qede/Makefile b/drivers/net/qede/Makefile
index 47e01be..884e173 100644
--- a/drivers/net/qede/Makefile
+++ b/drivers/net/qede/Makefile
@@ -47,6 +47,9 @@ CFLAGS_BASE_DRIVER += -Wno-unused-but-set-variable
 CFLAGS_BASE_DRIVER += -Wno-missing-declarations
 CFLAGS_BASE_DRIVER += -Wno-maybe-uninitialized
 CFLAGS_BASE_DRIVER += -Wno-strict-prototypes
+ifeq ($(shell test $(GCC_VERSION) -ge 60 && echo 1), 1)
+CFLAGS_BASE_DRIVER += -Wno-shift-negative-value
+endif
 else ifeq ($(CC), clang)
 CFLAGS_BASE_DRIVER += -Wno-format-extra-args
 CFLAGS_BASE_DRIVER += -Wno-visibility
-- 
2.5.5

